### PR TITLE
Add dxDiagram demo descriptions

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: JSDemos/Demos/Diagram/CustomShapesWithTemplatesWithEditing/description.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
-In this demo, the [customShapeTemplate](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#customShapeTemplate) option defines a common shape template and adds the **Edit** and **Delete** links to a shape. These links allow users to modify and remove employee data in the data source. The [reloadContent](/Documentation/ApiReference/UI_Widgets/dxDiagram/Methods/#reloadContentitemKey_applyLayout) method is called on every data source change. This method reloads the modified diagram elements from the data source and recreates the [diagram layout](/Documentation/Guide/Widgets/Diagram/Data_Binding/#Diagram_Layout).
+In this demo, the [customShapeTemplate](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#customShapeTemplate) option defines a common shape template and adds the **Edit** and **Delete** links to a shape. These links allow users to modify and remove employee data in the data source. The **Diagram** widget reloads the modified diagram data on every data source change.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Diagram/CustomShapesWithTemplatesWithEditing/description.md]:
```suggestion
In this demo, the [customShapeTemplate](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#customShapeTemplate) option defines a common shape template and adds the **Edit** and **Delete** links to a shape. These links allow users to modify and remove employee data in the data source. The **Diagram** widget reloads the modified diagram data every time the data source changes.

```

File: JSDemos/Demos/Diagram/CustomShapesWithTemplatesWithEditing/description.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
+In this demo, the [customShapeTemplate](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#customShapeTemplate) option defines a common shape template and adds the **Edit** and **Delete** links to a shape. These links allow users to modify and remove employee data in the data source. The **Diagram** widget reloads the modified diagram data on every data source change.
+
+The [onRequestLayoutUpdate](Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#onRequestLayoutUpdate) function specifies whether the widget should reapply the auto layout after the diagram is reloaded.
 
 The [customDataExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/#customDataExpr) option links custom employee information from the data source to diagram nodes. Changes in the data are reflected in the diagram history. Users can utilize the UI to undo and redo these changes.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Diagram/CustomShapesWithTemplatesWithEditing/description.md]:
```suggestion
The [customDataExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/#customDataExpr) option links custom employee information from the data source to diagram nodes. Changes made to the data are reflected in the diagram history. The undo and redo actions in the UI allow users to revert these changes.

```

File: JSDemos/Demos/Diagram/OperationRestrictions/description.md [hunk 3]
```diff
@@ -1 +1,3 @@
-TODO
\ No newline at end of file
+The **Diagram** widget allows you to restrict edit operations. To prohibit an operation, set the corresponding option in the [editing](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/editing/) property to **false**.
+
+In this demo, the [onRequestEditOperation](Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#onRequestEditOperation) function implements a complex logic to determine whether an operation is allowed. The **updateUI** event parameter indicates whether the event is raised by user actions or by the UI update. If a user try to perform a prohibited action, an error message is displayed.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Diagram/OperationRestrictions/description.md]:
```suggestion
In this demo, the [onRequestEditOperation](Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/#onRequestEditOperation) function implements a complex logic to determine whether an operation is allowed. The **updateUI** event parameter indicates whether the event is raised by user actions or UI update. If a user tries to perform a prohibited action, an error message is displayed.
```

File: JSDemos/Demos/Diagram/WebAPIService/description.md [hunk 4]
```diff
@@ -1 +1 @@
-This demo shows the **Diagram** widget loading data from a Web API service. 
+This example demonstrates the **Diagram** widget communicating with a Web API service. All data editing and data processing operations are delegated to the server. The <a href="https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data" target="_blank">DevExtreme.AspNet.Data</a> extension’s <a href="https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference" target="_blank">createStore</a> method configures access to the server from the client. The server implementation is available under the _DiagramEmployeesController.cs_ tab in the ASP.NET MVC version of this demo.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Diagram/WebAPIService/description.md]: "the **Diagram** widget communicating with a Web API service" Explain. How can you demonstrate 'communicating'?

Comment (dirk-pieterse) [reply JSDemos/Demos/Diagram/WebAPIService/description.md]: "All data editing and data processing operations are delegated to the server" -> All data editing and processing operations are delegated to the server

**Review by dirk-pieterse — DISMISSED** [review]

**Review by Romzay — APPROVED** [review]

**Pull request merged**